.gitlab/llnl-ci.yml

```diff
@@ -1,79 +1,132 @@
 stages:
+  - setup
+  - compile
   - update
-  - build rhel7-broadwell
-  - build rhel7-ppc64le
-  - build rhel7-cascadelake
+  - build
   - test
+  - cleanup
 
-Spack Update:
-  stage: update
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks $SPACK_WORK_DIR/spack || true
+    - chmod -fR 02770 $SPACK_WORK_DIR/spack    || true
+
+spack-setup:
+  stage: setup
   tags:
     - quartz
     - shell
   script:
-    - cd $SPACK_WORK_DIR/spack/
-    - git checkout HEAD^ .
-    - git pull --ff
-    - chmod -fR 02770 . || true
+    - mkdir -p $SPACK_WORK_DIR
+    - test -d $SPACK_WORK_DIR/spack && exit 0
```

Comment:
If the directory exists, it's assumed to be the repo, so quit?

Just making sure I understand. I assume the sequence of commands here does not abort on the first error?

Reply:
correct, this stage is responsible to clone the spack repo, automates the cloning process if need to run the pipeline after removing the spack dir